internal/pkg/version/version.go

```diff
@@ -4,65 +4,110 @@
 package version
 
 import (
+	"bytes"
 	"fmt"
-	"strings"
+	"time"
 )
 
 var (
+	// BuildDate is the time of the git commit used to build the program,
+	// in RFC3339 format. It is filled in by the compiler via makefile.
+	BuildDate string
+
 	// GitCommit and GitDescribe are filled by the compiler using ldflags to
 	// provide useful Git information.
 	GitCommit   string
 	GitDescribe string
 
 	// Version is the semantic version number describing the current state of
-	// NOM.
+	// nomad-pack.
 	Version = "0.0.1"
 
 	// Prerelease designates whether the current version is within a prerelease
 	// phase. Typically, this will be "dev" to signify a development cycle or a
 	// release candidate phase such as alpha, beta.1, rc.1, or such.
-	Prerelease = "techpreview.4"
+	VersionPrerelease = "techpreview.4"
 
 	// Metadata allows us to provide additional metadata information to the
 	// version identifier. This is typically used to identify enterprise builds
 	// using the "ent" metadata string.
-	Metadata = ""
+	VersionMetadata = ""
 )
 
-// HumanVersion composes the parts of the version in a way that's suitable for
-// displaying to humans.
-func HumanVersion() string {
-	version := Version
-	release := Prerelease
+// VersionInfo
+type VersionInfo struct {
+	BuildDate         time.Time
+	Revision          string
+	Version           string
+	VersionPrerelease string
+	VersionMetadata   string
+}
+
+func (v *VersionInfo) Copy() *VersionInfo {
+	if v == nil {
+		return nil
+	}
+
+	nv := *v
+	return &nv
+}
 
+func GetVersion() *VersionInfo {
+	ver := Version
+	rel := VersionPrerelease
+	md := VersionMetadata
 	if GitDescribe != "" {
-		version = GitDescribe
-	} else {
-		if release == "" {
-			release = "dev"
-		}
-
-		if release != "" && !strings.HasSuffix(version, "-"+release) {
-			// if we tagged a prerelease version then the release is in the version
-			// already.
-			version += fmt.Sprintf("-%s", release)
-		}
-
-		if Metadata != "" {
-			version += fmt.Sprintf("+%s", Metadata)
-		}
+		ver = GitDescribe
+	}
+	if GitDescribe == "" && rel == "" && VersionPrerelease != "" {
```

Comment:
I don't really want to open up a can of `build.yml` worms, because I don't think this _actually_ matters, but `GitDescribe` doesn't seem useful?  [in build.yml](https://github.com/hashicorp/nomad-pack/blob/cbf90695c94ad108ad59c6eed3e23c9f6e91468e/.github/workflows/build.yml#L69) it just gets set to a previous job's `product-version` output, which is the same (via `make version` => `scripts/version.sh`) as `VersionNumber()` in this very file?  seems very circuitous...

but maybe we just ignore it, because there seems like an awful lot of cruft in build.yml, so it could be full of refactorable rabbit holes, and it does work fine enough. ¯\\\_(ツ)\_/¯